### ISSUE TITLE
Publishing configurations are not shown in menu

### ISSUE THREAD

**Issue by kamil7x** [Komodo/KomodoEdit#49]
I added some configurations for publishing.
When I open right menu (with 3-bars icon) and choose Tools->Publishing there are no saved configurations.

In top menu (open with Alt) Tools->Publishing shows configurations properly

**Comment by cgchoffman** [comment]
Please include the following:
- OS type and version
- Version of Komodo (Help > About Komodo)
- Logs (Help > Troubleshooting > View Log File

**Comment by kamil7x** [comment]
Right, I forgot

OS: elementary OS 0.3 Freya (based on Ubuntu 14.04 LTS Trusty)
Komodo: Komodo IDE, version 9.0.0, build 87165, platform linux-x86_64. Built on Wed 
Log:
```
[2015-03-24 18:53:31,581] [INFO] Startup: Welcome to Komodo IDE 9.0.0 build 87165 (platform linux-x86_64, running on Linux 3.16.0-33-generic version #44~14.04.1-Ubuntu SMP Fri Mar 13 10:33:29 UTC 2015)
[2015-03-24 18:53:31,581] [INFO] Startup: /usr/bin/python built on Wed Mar 18 21:57:09 2015
[2015-03-24 18:53:32,039] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in resource://gre/modules/Preferences.jsm:381
[2015-03-24 18:53:32,041] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'component' directive. (1) in file:///mnt/dane/programy/Komodo-IDE/lib/mozilla/extensions/scope-files@activestate.com/components/component.manifest:1
[2015-03-24 18:53:32,042] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'contract' directive. (1) in file:///mnt/dane/programy/Komodo-IDE/lib/mozilla/extensions/scope-files@activestate.com/components/component.manifest:2
[2015-03-24 18:53:32,044] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'interfaces' directive. (1) in file:///mnt/dane/programy/Komodo-IDE/lib/mozilla/extensions/scope-files@activestate.com/components/component.manifest:3
[2015-03-24 18:53:32,129] [WARNING] root: ko.logging has been converted to a CommonJS module; use require("ko/logging") instead (since Komodo 9.0.0a1).
    @chrome://komodo/content/komodo.js:15:1

[2015-03-24 18:53:32,556] [WARNING] console-logger: Unknown property 'border-radius-top-left'.  Declaration dropped. (1) in less://commando/skin/commando.less:18
[2015-03-24 18:53:32,557] [WARNING] console-logger: Unknown property 'border-radius-top-right'.  Declaration dropped. (1) in less://commando/skin/commando.less:19
[2015-03-24 18:53:33,035] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in file:///mnt/dane/programy/Komodo-IDE/lib/mozilla/components/koamAddonManager.js:138
[2015-03-24 18:53:33,863] [WARNING] koSkin: Detected GTK theme via libgtk: elementary
[2015-03-24 18:53:34,496] [WARNING] root: ko.findresults was already loaded, re-creating it.

[2015-03-24 18:53:34,901] [DEBUG] places_js: waiting for document complete
[2015-03-24 18:54:08,742] [WARNING] console-logger: Key event not available on GTK2: key="u" modifiers="control,shift" (1) in chrome://komodo/content/komodo.xul:0
[2015-03-24 18:54:08,743] [WARNING] console-logger: Key event not available on some keyboard layouts: key="u" modifiers="control,alt" (1) in chrome://komodo/content/komodo.xul:0
[2015-03-24 18:54:08,744] [WARNING] console-logger: Key event not available on some keyboard layouts: key="v" modifiers="control,alt" (1) in chrome://komodo/content/komodo.xul:0
[2015-03-24 18:55:50,200] [WARNING] console-logger: Key event not available on GTK2: key="u" modifiers="control,shift" (1) in chrome://komodo/content/tail/tail.xul:0
[2015-03-24 18:55:50,200] [WARNING] console-logger: Key event not available on some keyboard layouts: key="u" modifiers="control,alt" (1) in chrome://komodo/content/tail/tail.xul:0
[2015-03-24 18:55:50,201] [WARNING] console-logger: Key event not available on some keyboard layouts: key="v" modifiers="control,alt" (1) in chrome://komodo/content/tail/tail.xul:0
```

**Comment by jellisii** [comment]
This is not just Linux only.  Confirmed on Windows 8.1 64b.  

Komodo IDE, version 9.0.0, build 87165, platform win32-x86.
Built on Wed Mar 18 21:58:08 2015.
```
[2015-03-25 10:00:55,375] [INFO] Startup: Welcome to Komodo IDE 9.0.0 build 87165 (platform win32-x86, running on Windows post2012Server version 6.3.9600)
[2015-03-25 10:00:55,375] [INFO] Startup: C:\Program Files (x86)\ActiveState Komodo IDE 9\lib\mozilla\komodo.exe built on Wed Mar 18 21:58:08 2015
[2015-03-25 10:00:55,970] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in resource://gre/modules/Preferences.jsm:381
[2015-03-25 10:00:55,971] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'component' directive. (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%209/lib/mozilla/extensions/scope-files@activestate.com/components/component.manifest:1
[2015-03-25 10:00:55,971] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'contract' directive. (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%209/lib/mozilla/extensions/scope-files@activestate.com/components/component.manifest:2
[2015-03-25 10:00:55,973] [WARNING] console-logger: Bootstrapped manifest not allowed to use 'interfaces' directive. (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%209/lib/mozilla/extensions/scope-files@activestate.com/components/component.manifest:3
[2015-03-25 10:00:56,025] [WARNING] root: ko.logging has been converted to a CommonJS module; use require("ko/logging") instead (since Komodo 9.0.0a1).
    @chrome://komodo/content/komodo.js:15:1

[2015-03-25 10:00:56,326] [WARNING] console-logger: Unknown property 'border-radius-top-left'.  Declaration dropped. (1) in less://commando/skin/commando.less:68
[2015-03-25 10:00:56,328] [WARNING] console-logger: Unknown property 'border-radius-top-right'.  Declaration dropped. (1) in less://commando/skin/commando.less:69
[2015-03-25 10:00:56,680] [ERROR] console-logger: bad script XDR magic number (0) in :0
Traceback (most recent call last):
  File "", line 0, in 

[2015-03-25 10:00:56,831] [WARNING] keybindings: [Ctrl+Shift+U] was used for 'cmd_convertLowerCase', overriding to use 'cmd_scope-sections'
[2015-03-25 10:00:59,104] [WARNING] root: ko.findresults was already loaded, re-creating it.

[2015-03-25 10:00:59,180] [DEBUG] places_js: waiting for document complete
[2015-03-25 10:01:01,380] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%209/lib/mozilla/components/koamAddonManager.js:138
[2015-03-25 10:01:11,582] [ERROR] koRemoteTransfer: koTransferUploadOperation:: unexpected exception
Traceback (most recent call last):
  File "C:\Program Files (x86)\ActiveState Komodo IDE 9\lib\mozilla\extensions\publishing@ActiveState.com\components\koRemoteTransfer.py", line 361, in run
    dirname(localpath))
  File "C:\Program Files (x86)\ActiveState Komodo IDE 9\lib\mozilla\extensions\publishing@ActiveState.com\components\koRemoteTransfer.py", line 272, in _createRemoteDirsFromLocalDirs
    orig_remote_uri, ))
Exception: Could not find any parent directory for u'file:///var/www/demo-manager/app/controllers'
[2015-03-25 10:02:09,598] [WARNING] console-logger: Key event not available on GTK2: key="u" modifiers="control,shift" (1) in chrome://komodo/content/komodo.xul:0
[2015-03-25 10:02:09,601] [WARNING] console-logger: Key event not available on GTK2: key="u" modifiers="control,shift" (1) in chrome://komodo/content/komodo.xul:0
[2015-03-25 10:02:09,601] [WARNING] console-logger: Key event not available on some keyboard layouts: key="v" modifiers="control,alt" (1) in chrome://komodo/content/komodo.xul:0
[2015-03-25 10:05:49,207] [WARNING] console-logger: Key event not available on GTK2: key="u" modifiers="control,shift" (1) in chrome://komodo/content/tail/tail.xul:0
[2015-03-25 10:05:49,207] [WARNING] console-logger: Key event not available on GTK2: key="u" modifiers="control,shift" (1) in chrome://komodo/content/tail/tail.xul:0
[2015-03-25 10:05:49,207] [WARNING] console-logger: Key event not available on some keyboard layouts: key="v" modifiers="control,alt" (1) in chrome://komodo/content/tail/tail.xul:0
```

**Comment by Naatan** [comment]
@cgchoffman were you able to reproduce this?

**Comment by cgchoffman** [comment]
I haven't seen this no.  I've been playing with Publishing quite a bit resolve issue #40 as well so I thought I would have hit it by now.

**Comment by Naatan** [comment]
For those reproducing the bug - were you upgrading from Komodo 8? Are the configs that are missing those you created in Komodo 8?

**Comment by kamil7x** [comment]
I am not sure for now, but it is possible, that configs were from Komodo 8

**Comment by Naatan** [comment]
Do you have a "Publishing" entry when you right click a file in the Places widget? (Left side pane).

**Comment by kamil7x** [comment]
Yes, i have:

Publishing 
|_ Push
|_ Diff

**Comment by cgchoffman** [comment]
Can one of you send your prefs.xml file and the contents of the publishing folder in your Komodo profile to me (careyh@activestate.com) through email?

You can find your profile folder with this [Komodo Files Location FAQ](https://community.activestate.com/faq/komodo-file-locations#appdata_dir)

**Comment by kamil7x** [comment]
sorry, I hadn't much time for testing and other stuff last weeks

@cgchoffman did @jellisii send you that folder? if not I'll send you it in the evening

**Comment by cgchoffman** [comment]
I do'nt believe I got anything.  I don't recall an email about publishing configurations.  If you could send that, that would be helpful.  Thank you.

**Comment by rb40** [comment]
Hi there

I can confirm this is an issue on Windows. I upgraded from V8 to V9 and all my publishing configuration are no longer accessible, the UI is acting as if they never existed. If I add a new publishing configuration it is not shown in the menu as soon as you close the publishing window. I've taken a look at my prefs.xml file and the records appear to be in there but they are not being shown in the menu.

Thanks

Richard

**Comment by rb40** [comment]
@cgchoffman can you confirm you have enough information to debug this issue as it is causing me a real headache.

Thanks

Richard

**Comment by cgchoffman** [comment]
I have the email from Kamil.  I have not been able to look into this issue since getting it though.  I've made this priority for 9.1 so hopefully we can get it in.

**Comment by Naatan** [comment]
Able to reproduce now. Will look into.

**Comment by cgchoffman** [comment]
I'm still not able to repro using @kamil7x publishing configs.  They load properly.  @kamil7x , could you try what rb40 (if you haven't already) and create a new publishing config and see if that jump starts the menu items to reload?

Another workaround might be to delete existing publishing configs that you have, move the publishing folder in your profile folder aside, then create a new publishing config.

**Comment by Naatan** [comment]
Fairly certain this was fixed in 9.1
